ui/typography.ts

```diff
@@ -1,97 +1,116 @@
+  button = "button",
+}
 
-const H2 = css`
-  font-family: "Inter";
-  font-weight: 700;
-  font-size: 64px;
-  line-height: 77px;
-  letter-spacing: -0.02em;
-  ${media.PHONE} {
-    font-size: 36px;
-    line-height: 44px;
-  }
-`;
+interface ITypography {
+  color?: string;
+  variant: VariantsTypography;
+}
 
-const H3 = css`
-  font-family: "Inter";
-  font-weight: 700;
-  font-size: 36px;
-  line-height: 44px;
-`;
-const H5 = css`
-  font-family: "Inter";
-  font-weight: 700;
-  font-size: 28px;
-  line-height: 36px;
-`;
+const variantTypography = (variant = VariantsTypography.subtitle) =>
+  ({
+    h1: css`
+      font-family: "Inter";
+      font-weight: 700;
+      font-size: 56px;
+      line-height: 64px;
+      ${media.PHONE} {
+        font-size: 32px;
+        line-height: 44px;
+      }
+    `,
+    h2: css`
+      font-family: "Inter";
+      font-weight: 700;
+      font-size: 64px;
+      line-height: 77px;
+      letter-spacing: -0.02em;
 
-const subtitle = css`
-  font-family: "Inter";
-  font-weight: 400;
-  font-size: 18px;
-  line-height: 30px;
-  letter-spacing: -0.02em;
-`;
+      ${media.TABLET} {
+        font-size: 54px;
+        line-height: 65px;
+      }
+      ${media.PHONE} {
+        font-size: 36px;
+        line-height: 44px;
+      }
+    `,
+    h3: css`
+      font-family: "Inter";
+      font-weight: 700;
+      font-size: 36px;
+      line-height: 44px;
+    `,
+    h5: css`
+      font-family: "Inter";
+      font-weight: 700;
+      font-size: 28px;
+      line-height: 36px;
+    `,
+    paragraphSmall: css`
+      font-family: "Inter";
+      font-weight: 400;
+      font-size: 18px;
+      line-height: 30px;
+    `,
+    paragraphXS: css`
+      font-family: "Inter";
+      font-weight: 400;
+      font-size: 16px;
+      line-height: 20px;
+    `,
+    paragraphSmallBold: css`
+      font-family: "Inter";
+      font-weight: 700;
+      font-size: 18px;
+      line-height: 30px;
+    `,
+    paragraphMedium: css`
+      font-family: "Inter";
+      font-weight: 400;
+      font-size: 24px;
+      line-height: 36px;
+    `,
+    paragraphMediumBold: css`
+      font-family: "Inter";
+      font-weight: 400;
+      font-size: 24px;
+      line-height: 36px;
+    `,
+    subtitle: css`
+      font-family: "Inter";
+      font-weight: 400;
+      font-size: 18px;
+      line-height: 30px;
+    `,
+    button: css`
+      font-family: "Inter";
+      font-weight: 500;
+      font-size: 18px;
+      line-height: 23px;
+      ${media.PHONE} {
+        font-weight: 400;
+        font-size: 18px;
+        line-height: 30px;
+      }
+    `,
+  }[variant]);
 
-const bodytext = css`
-  font-family: "Inter";
-  font-weight: 400;
-  font-size: 16px;
-  line-height: 20px;
-  letter-spacing: -0.02em;
-`;
-const paragraph1 = css`
-  font-family: "Inter";
-  font-weight: 600;
-  font-size: 24px;
-  line-height: 36px;
+export const Typography = styled.p<ITypography>`
+  ${({ variant }) => variantTypography(variant)}
```

Comment:
you duplicate one css styles in every variants, maybe will better to move it to general Typography styles?